Makefile

```diff
@@ -55,7 +55,9 @@ help: default
 .PHONY: deps
 deps: default
 ifeq ($(LUAROCKS_VER),luarocks 3.)
+ifeq ("$(wildcard ~/.luarocks)", "")
```

Comment:
I think calling `mkdir ~/.luarocks` even it exists doesn't do any harm?

Reply:
It would leave an error message, which confused me at first before I realize it does no harm

Reply:
@Yiyiyimu 
I think you are making something wrong. The `make deps` only install dependencies locally. There is no need to call `sudo`.

Reply:
Got it I installed luarocks with `apt`, so when it changes luarocks config, it needs sudo privilege. I'll change to the method mentioned in #3467 and see if it works.

Reply:
Use `mkdir -p ~/.luarocks` would be better.

Reply:
Yeah that would be better! Thanks!

And I made a mistake yesterday. No matter how we install luarocks, the config would always write to `/usr/local/etc/luarocks`, so we all need sudo permission. Is that the same on your side?

Reply:
What about:
```diff
 deps: default
 ifeq ($(LUAROCKS_VER),luarocks 3.)
        mkdir ~/.luarocks || true
-       luarocks config variables.OPENSSL_LIBDIR $(addprefix $(OR_PREFIX), openssl/lib)
+       luarocks config --local variables.OPENSSL_LIBDIR $(addprefix $(OR_PREFIX), openssl/lib)
        luarocks config variables.OPENSSL_INCDIR $(addprefix $(OR_PREFIX), openssl/include)
        luarocks install rockspec/apisix-master-0.rockspec --tree=deps --only-deps --local
 else
```

Reply:
It's weird... I installed lua5.3 and configure luarocks to it this morning to do some stuff, and found out it changes some config  that even though I re-configure luarocks to luajit from openresty, it still tries to find lua5.3.

So I remove everything about luarocks, and re-install it with the script. Now this time, it would write config to `~/.luarocks` without the `--local` flag.

I'm not so sure right now. If you could not reproduce this problem, I'll close it